x-pack/filebeat/input/httpjson/config/config_oauth.go

```diff
@@ -67,7 +67,8 @@ func (o *OAuth2) IsEnabled() bool {
 
 // Client wraps the given http.Client and returns a new one that will use the oauth authentication.
 func (o *OAuth2) Client(ctx context.Context, client *http.Client) (*http.Client, error) {
```

Comment:
Where do we need this type? If the implementation is internal to the httpjson package, why not keep it in the httpjson package and unexport it (implementation detail). If this type (and corresponding types/values) are more generic, consider to move it to the `libbeat/common/transport` package (or a sub-package).